tests/spread/plugins/v1/x-local/local-plugin/task.yaml

```diff
@@ -9,7 +9,12 @@ prepare: |
   . "$TOOLS_DIR/snapcraft-yaml.sh"
   set_base "$SNAP_DIR/snap/snapcraft.yaml"
 
+  deb_arch=$(dpkg --print-architecture)
+  sed -i state-pull -e"s/\(.*deb_arch:\) @EXPECTED_ARCHITECTURE@$/\1 $deb_arch/"
```

Comment:
you're method is good, though I would have gone with a simpler:
`sed -i state-pull "s|@EXPECTED_ARCHITECTURE@|$deb_arch|"`